### PR TITLE
Implemented collidesAt and wouldCollideUsing

### PR DESCRIPTION
**Pull request by 4everTheOne**
# Description
Implements the collision detection in entity.

# Checklist
The following items should be checked before the pull request can be merged.
- [X] Code follows existing style.
- [X] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java [hunk 1]
```diff
@@ -1399,15 +1399,13 @@ public boolean isInPowderedSnow()
 	@Override
 	public boolean collidesAt(@NotNull Location location)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:1403]: I think the paper implementation has some more details to it. For example if entity is instance of armorstand and this paper config property is false `entities.armorStands.doCollisionEntityLookups`, return false. (Do we even have an paper config mock?)

I think there's some more things to this, but I honestly can't comprehend it, seems tideous to mock properly. Something something about checking against blocks and world border.

File: src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java [hunk 2]
```diff
@@ -1399,15 +1399,13 @@ public boolean isInPowderedSnow()
...
+		return getBoundingBox().contains(location.toVector());
 	}
 
 	@Override
 	public boolean wouldCollideUsing(@NotNull BoundingBox boundingBox)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return getBoundingBox().overlaps(boundingBox);
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:1408]: Same can be said about this implementation as with the previous comment

File: src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java (hunk 1 above)
Comment (4everTheOne) [reply src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:1403]: That's correct, but at the moment we don't have a mock for _PaperConfigurations_, only for _ServerConfiguration_.
How should we address this?

Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:1403]: > That's correct, but at the moment we don't have a mock for _PaperConfigurations_, only for _ServerConfiguration_. How should we address this?

Let's  not address this then // just ignore that behavior.

**Review by Thorinwasher — APPROVED** [review]

**Pull request merged**